WikiDrips/UImageExtensions.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  TitleImageView.swift
+//  WikiDrips
+//
+//  Created by Bart Cullimore on 4/6/17.
+//  Copyright © 2017 Bart Cullimore. All rights reserved.
+//
+
+import UIKit
+
+extension UIImage {
+    
+    func image(withInitials: String?, in bounds: CGRect) -> UIImage {
+        let scale = UIScreen.main.scale
+        
+        UIGraphicsBeginImageContextWithOptions(bounds.size, false, scale)
+        defer {
+            UIGraphicsEndImageContext()
+        }
+        
+        guard let initials = withInitials, let context = UIGraphicsGetCurrentContext() else {
+            return #imageLiteral(resourceName: "PlaceHolderImage")
+        }
+
+        let circlePath = CGPath(ellipseIn: bounds, transform: nil)
+        context.addPath(circlePath)
+        context.clip()
+        
+        UIColor.black.setFill()
+        context.fill(bounds)
+        
+        let textAttrs: [String : AnyObject] = [NSForegroundColorAttributeName : UIColor.white]
+        let textRectSize = initials.size(attributes: textAttrs)
+        let textRect = CGRect(x: bounds.size.width / 2 - textRectSize.width / 2,
```

Comment:
This is fine, but I might have written it as: 
```
let textRect =  CGRect(x: bounds.midX - textRectSize.width / 2,
                                  y: bounds.midY - textRectSize.height / 2,
                                  width: textRectSize.width,
                                  height: textRectSize.height).integral
```
There's some helpful CGRect properties that you should know about. `min/max/mid X/Y`. Also `integral` is good here because if text is on a fractional pixel it can look blurry.